test/cli/cli_test.hpp

```diff
@@ -0,0 +1,92 @@
+#include <gtest/gtest.h>
+
+#include <seqan3/test/expect_range_eq.hpp>
+#include <cstdlib>               // system calls
+#include <sstream>               // ostringstream
+#include <string>                // strings
+#include <filesystem>
+
+#pragma once
+
+// Provides functions for CLI test implementation.
+struct cli_test : public ::testing::Test
+{
+private:
+
+    // Holds the original work directory where Gtest has been started.
+    std::filesystem::path original_workdir{};
+
+protected:
+
+    // Result struct for captured streams and exit code.
+    struct cli_test_result
+    {
+        std::string out{};
+        std::string err{};
+        int exit_code{};
+    };
+
+    // Invoke the app execution. The command line call should be given as separate parameters.
+    template <typename... CommandItemTypes>
+    cli_test_result execute_app(CommandItemTypes &&... command_items)
+    {
+        cli_test_result result{};
+
+        // Assemble the command string and disable version check.
+        std::ostringstream command{};
+        command << "SEQAN3_NO_VERSION_CHECK=1 " << BINDIR;
+        int a[] = {0, ((void)(command << command_items << ' '), 0) ... };
+        (void) a;
```

Comment:
I think the following should work as well and be more readable:
```suggestion
        ((command << command_items << ' '), ...);
```

Reply:
I would actually not use the stringstream at all, and just use `std::string` and ` +=` but I don't care much about this.